src/internal/nft-api-unichain.ts

```diff
@@ -0,0 +1,155 @@
+import { AlchemyConfig } from '../api/alchemy-config';
+import {
+  GetBaseNftsForOwnerUnichainOptions,
+  GetNftsForOwnerUnichainOptions,
+  Network,
+  OwnedBaseNftsResponse,
+  OwnedBaseNftsResponseUnichain,
+  OwnedNftsResponse,
+  OwnedNftsResponseUnichain
+} from '../types/types';
+import { getNftsForOwner } from './nft-api';
+import { NetworkPageKey, UnichainPageKeyCache } from './page-key';
+
+function networksAreEqual(a: Network[], b: Network[]): boolean {
+  return JSON.stringify(a.sort()) === JSON.stringify(b.sort());
+}
+
+function validateParams(
+  networks: Network[],
+  options:
+    | GetNftsForOwnerUnichainOptions
+    | GetBaseNftsForOwnerUnichainOptions = {}
+): {
+  unichainPageKeyCache: UnichainPageKeyCache;
+  networkPageKeys: Map<Network, NetworkPageKey>;
+} {
+  // We could default to all networks, but as we add or remove support for
+  // various networks that list can change. We force developers to explicitly
+  // list their desired networks so that their code is future-proof.
+  if (!networks || networks.length === 0) {
+    throw new Error(
+      'Must provide a `networks` parameter for unichain requests.'
+    );
+  }
+
+  const { unichainPageKeyCache, pageKey = null } = options;
+
+  // This should be provided by default in the namespace, so realistically
+  // should never happen. In non-unichain methods we do not use a page key
+  // cache because page keys are handled by the API. However we have no API
+  // for unichain, and so the SDK needs to handle page keys.
+  if (!unichainPageKeyCache) {
+    throw new Error('No page key cache was provided.');
+  }
+
+  const networkPageKeys = unichainPageKeyCache.getNetworkPageKeys(pageKey);
+
+  // If the user made an initial request for a set of networks, we do not
+  // allow them to change networks on the next page. We could technically
+  // allow this, but it would require significantly more magic.
+  const originalNetworks = Array.from(networkPageKeys.keys());
+  if (pageKey && !networksAreEqual(originalNetworks, networks)) {
+    throw new Error('Networks must not change between pages.');
+  }
+
+  return {
+    unichainPageKeyCache,
+    networkPageKeys
+  };
+}
+
+function buildResponseWithPageKey(
+  networks: Network[],
+  networkResults: (OwnedNftsResponse | OwnedBaseNftsResponse)[],
+  unichainPageKeyCache: UnichainPageKeyCache
+) {
+  const response: OwnedNftsResponseUnichain = {
+    nftsByNetwork: new Map()
+  };
+  const networkPageKeys = new Map();
```

Comment:
Should help out the type system and readers a little here, either
```ts
const networkPageKeys: NetworkPageKeys = new Map();
```
or
```ts
const networkPageKeys = new Map<Network, NetworkPageKey>();
```
Otherwise, the type system can't infer what kind of `Map` this is.